packages/solid-query/src/createBaseQuery.ts

```diff
@@ -45,40 +49,100 @@ export function createBaseQuery<
     observer.getOptimisticResult(defaultedOptions),
   )
 
-  const [dataResource, { refetch, mutate }] = createResource<TData | undefined>(
+  const createServerSubscriber = (
+    resolve: (
+      data:
+        | QueryObserverResult<TData, TError>
+        | PromiseLike<QueryObserverResult<TData, TError> | undefined>
+        | undefined,
+    ) => void,
+  ) => {
+    return observer.subscribe((result) => {
+      notifyManager.batchCalls(() => {
+        const unwrappedResult = { ...unwrap(result) }
+        setState(unwrappedResult)
+        resolve(unwrappedResult)
+      })()
+    })
+  }
+
+  const createClientSubscriber = (refetch: (info?: unknown) => void) => {
+    return observer.subscribe((result) => {
+      notifyManager.batchCalls(() => {
+        const unwrappedResult = { ...unwrap(result) }
+        setState(unwrappedResult)
+        refetch()
+      })()
+    })
+  }
```

Comment:
```suggestion
  const createClientSubscriber = (refetch: (info?: unknown) => void, hydrated: boolean = false) => {
    if (hydrated) {
        defaultedOptions.refetchOnMount = false
        observer.setOptions(defaultedOptions)
     }
    return observer.subscribe((result) => {
      notifyManager.batchCalls(() => {
        const unwrappedResult = { ...unwrap(result) }
        setState(unwrappedResult)
        refetch()
      })()
    })
  }
```

Can we also add this to the client subscriber? This would allow us to not refetch a query that was rendered on the server, even if a staleTime wasn't set. Similar to how CSR apps behave

Reply:
For sure - I'll give it a try now. Would also be nice if we could figure out an actual "ssr" test that covers that hydrate code branch, and this stale behavior. Will see if I can figure that out.

Reply:
👍 done in 9fc9e3d7142b2b812f4950b678aad6b080c7fc03. Haven't figured out the ssr test... was looking for inspiration, but can't even find tests in the solidjs repo that cover ssr related functionality.

Reply:
Great work! I can add the tests for SSR in a separate PR. Could you resolve the conflicts before I run the CI workflow?

Reply:
👍 done